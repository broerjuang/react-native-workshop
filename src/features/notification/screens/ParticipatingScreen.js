// @flow
import React, {Component} from 'react';
import {View, Text, StyleSheet} from 'react-native';

export default class ParticipatingScreen extends Component<{}> {
  render() {
    const {noNotifText, mainContainer} = styles;

    return (
      <View style={mainContainer}>
        <Text style={noNotifText}>
          You dont have any notifications of this type.
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: 'white',
  },
  noNotifText: {
    marginTop: 10,
    textAlign: 'center',
    alignSelf: 'center',
  },
  mainList: {
    height: 100,
    margin: 10,
    shadowOffset: {width: 1, height: 1},
    shadowColor: 'black',
    shadowOpacity: 0.2,
  },
  upperListPart: {
    flex: 1,
    backgroundColor: '#F1F1F1',
  },
  listDetails: {
    flexDirection: 'row',
    margin: 10,
  },
  listText: {
    marginLeft: 5,
    alignSelf: 'center',
    fontWeight: 'bold',
  },
  bottomListPart: {
    flexDirection: 'row',
    flex: 1,
    backgroundColor: 'white',
    paddingLeft: 5,
  },
  bottomDetailsText: {
    textAlign: 'left',
    alignSelf: 'center',
    padding: 5,
    flexWrap: 'wrap',
    fontSize: 12,
  },
});
