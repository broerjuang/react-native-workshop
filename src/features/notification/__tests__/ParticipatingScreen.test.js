// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import ParticipatingScreen from '../screens/ParticipatingScreen';

describe('ParticipatingScreen', () => {
  it('should render ParticipatingScreen corectly', () => {
    let component = renderer.create(
      <ParticipatingScreen navigation={{navigate: (s) => s}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
