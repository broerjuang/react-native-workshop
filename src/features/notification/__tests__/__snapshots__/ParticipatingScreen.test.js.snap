// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipatingScreen should render ParticipatingScreen corectly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    You dont have any notifications of this type.
  </Text>
</View>
`;
